system-int-test/test/cmr/system_int_test/search/variable/concept_retrieval_test.clj

```diff
@@ -0,0 +1,99 @@
+(ns cmr.system-int-test.search.variable.concept-retrieval-test
+  "Integration test for variable retrieval via the /concepts/:concept-id and
+  /concepts/:concept-id/:revision-id endpoints."
+  (:require
+   [cheshire.core :as json]
+   [clojure.test :refer :all]
+   [cmr.common.mime-types :as mt]
+   [cmr.mock-echo.client.echo-util :as e]
+   [cmr.system-int-test.system :as s]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.metadata-db-util :as mdb]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.system-int-test.utils.variable-util :as variable]
+   [cmr.transmit.config :as transmit-config]))
+
+(use-fixtures
+ :each
+ (join-fixtures
+  [(ingest/reset-fixture {"provguid1" "PROV1"
+                          "provguid2" "PROV2"
+                          "provguid3" "PROV3"})
+   variable/grant-all-variable-fixture]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Retrieve by concept-id - general test
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(deftest retrieve-variable-by-concept-id
+  (let [var1-name "variable1"
+        var1-concept (variable/make-variable-concept {:Name var1-name})
+        var1 (ingest/ingest-concept var1-concept)
+        _ (index/wait-until-indexed)
+        var1-v2 (ingest/ingest-concept var1-concept)
```

Comment:
@chris-durbin: Question about this ... I've tested for the same concept-id but different revisions ... and your comment above implies that's not enough? Which leads some some confusion on my part ... (because I don't know enough about CMR concepts and revisioning, no doubt). Can you give me an example of where the concept-id + revision increment check would give a false positive, where a check against the updated name would not? This would help me greatly ...